# Demo Notebook for RC-Mixup
This notebook demonstrates how to use the RC-Mixup step by step.

In [1]:
# Import necessary modules
from model import *
from utils import *
from train import *
import numpy as np

## Step 1: Data Preprocessing

In [2]:
# Hyperparameters

label_dim = 4
noise_ratio = 0.3
device = torch.device('cuda')
seed = 410
warmup_epochs = 1000
update_epochs = 1000
lr_model = 1e-2
batch_size = 128
alpha = 2
L = 500
N = 500
bw_list = [5, 10, 15, 20]
start_bandwidth = 20

In [3]:
raw_data = np.load("Synthetic_dataset.npy")

dataset = {}
dataset[seed] = {}
train_data, valid_data, test_data, Y_max, Y_min, noise_idx = preprocess_dataset(seed, raw_data, noise_ratio)
x_tr, y_tr = train_data[:, :-label_dim], train_data[:, -label_dim:]
x_val, y_val = valid_data[:, :-label_dim], valid_data[:, -label_dim:]
x_test, y_test = test_data[:, :-label_dim], test_data[:, -label_dim:]

dataset[seed]['x_tr'], dataset[seed]['y_tr'] = x_tr, y_tr
dataset[seed]['x_val'], dataset[seed]['y_val'] = x_val, y_val
dataset[seed]['x_test'], dataset[seed]['y_test'] = x_test, y_test
dataset[seed]['Y_max'], dataset[seed]['Y_min'] = Y_max, Y_min
dataset[seed]['noise_idx'] = noise_idx

## Step 2: Model Training

In [4]:
updated_model = rc_mixup(dataset, noise_ratio, seed, lr_model, alpha, warmup_epochs, update_epochs,
             L, N, bw_list, start_bandwidth, batch_size, device = device)

Seed: 410,
Val RMSE: 7.2685, Val MAPE: 5.9069,
Test RMSE: 8.1799, Test MAPE: 6.1727,
Runtime: 622.3s


Seed: 410,
Val RMSE: 7.2685, Val MAPE: 5.9069,
Test RMSE: 8.1799, Test MAPE: 6.1727,
Runtime: 648.8s